examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,204 @@
+# Nvflare with TensorFlow
+
+This example uses Nvflare to train an image classifier using federated averaging and TensorFlow as the deep learning framework.
+
+[NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) (**NV**IDIA **F**ederated **L**earning **A**pplication **R**untime **E**nvironment)
+is a domain-agnostic, open-source, extensible SDK that allows researchers and data scientists to adapt existing ML/DL workflows to a federated paradigm.
+It enables platform developers to build a secure, privacy-preserving offering for a distributed multi-party collaboration.
+
+[Here](https://nvflare.readthedocs.io/en/main/flare_overview.html#high-level-system-architecture) is a high level system architecture of Nvflare.
+
+> **_NOTE:_** This example uses the [MNIST](http://yann.lecun.com/exdb/mnist/) handwritten digits dataset and will load its data within the trainer code.
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+## High-level diagram
+
+The following diagram shows one server and two clients that are connected to the server through a secure gRPC link:
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the Nvflare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## 1. Create custom image
+
+For this demo to work, you have to create a custom image with TensorFlow and Nvflare installed. To create the image and push it on Artifact Registry, go to the `docker-image` and build the image:
+
+```bash
+cd docker-image
+export REGION=$(gcloud config get compute/region)
+export PROJET_ID=$(gcloud config get core/project)
+export REPOSITORY=my-repository
+gcloud builds submit --tag ${REGION}-docker.pkg.dev/${PROJECT_ID}/${REPOSITORY}/nvflare-tensorflow:1.0.0
+```
+
+Once the image is built, modify the `kustomization.yaml` file to add the new name and new tag of the image.
+
+## 2. Create infrastructure to store the models
+
+All the models generated will be stored in a Cloud storage bucket mounted by each pod. To create the Cloud storage and give the right permissions, from the `terraform` folder, run :
```

Comment:
I wouldn't explain these details because we'd have to keep the up to date. I would write a more generic sentence, as you did with the title.